IndoorMapsBackend/src/resolvers/AreaResolver.ts

```diff
@@ -0,0 +1,140 @@
+import { InputType, Field, Int, Resolver, Mutation, Arg, Ctx, Query } from "type-graphql"
+import { getUserOrThrowError } from "../auth/validateUser.js"
+import { Context } from "../utils/context.js"
+import { Area, NewAreaResult } from "../graphqlSchemaTypes/Area.js"
+import { convertToGraphQlArea } from "../utils/typeConversions.js"
+import { GraphQLError } from "graphql"
+
+@InputType()
+class AreaCreateInput {
+    @Field(type => Int)
+    floorDatabseId: number
+
+    @Field()
+    title: string
+
+    @Field()
+    description: string
+
+    @Field()
+    shape: string
+
+    @Field({ nullable: true })
+    category: string
+}
+
+@InputType()
+class AreaUniqueInput {
+    @Field(type => Int)
+    id: number
+}
+
+@InputType()
+class AreaModifyInput extends AreaUniqueInput {
+    @Field({ nullable: true })
+    title?: string
+
+    @Field({ nullable: true })
+    description?: string
+
+    @Field({ nullable: true })
+    shape?: string
+
+    @Field({ nullable: true })
+    category?: string
+}
+
+@Resolver(of => Area)
+export class AreaResolver {
+    @Mutation((returns) => NewAreaResult)
+    async createArea(
+        @Arg('data') data: AreaCreateInput,
+        @Ctx() ctx: Context,
+    ): Promise<NewAreaResult> {
+        // TODO: check if user is alowed to edit this building
+        await getUserOrThrowError(ctx.cookies);
+        const newArea = await ctx.prisma.area.create({
+            data: {
+                title: data.title,
+                description: data.description,
+                shape: JSON.parse(data.shape),
```

Comment:
Since shape is an optional field, in case of empty strings, this would throw an error.

Reply:
Shape is not an optional Field for areas. In order to make the creation and deletion of areas feel snappy, they are created when a new shape is created and deleted when a new shape is delete.